src/Windows/Avalonia.Win32/Interop/Automation/IRawElementProviderSimple.cs

```diff
@@ -274,10 +275,130 @@ internal enum UiaControlTypeId
     [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
     public interface IRawElementProviderSimple
     {
+#if NET6_0_OR_GREATER
+        public static readonly Guid IID = new("d6dd68d1-86fd-4332-8666-9abedea2d24c");
+        public const int VtblSize = 3 + 4;
+#endif
         ProviderOptions ProviderOptions { get; }
         [return: MarshalAs(UnmanagedType.IUnknown)]
         object? GetPatternProvider(int patternId);
         object? GetPropertyValue(int propertyId);
         IRawElementProviderSimple? HostRawElementProvider { get; }
     }
+
+#if NET6_0_OR_GREATER
+    internal static unsafe class IRawElementProviderSimpleManagedWrapper
+    {
+        [UnmanagedCallersOnly]
+        public static int GetProviderOptions(void* @this, ProviderOptions* ret)
+        {
+            try
+            {
+                *ret = ComWrappers.ComInterfaceDispatch.GetInstance<IRawElementProviderSimple>((ComWrappers.ComInterfaceDispatch*)@this).ProviderOptions;
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                return ex.HResult;
+            }
+        }
+
+        [UnmanagedCallersOnly]
+        public static int GetPatternProvider(void* @this, int patternId, void** ret)
+        {
+            try
+            {
+                var obj = ComWrappers.ComInterfaceDispatch.GetInstance<IRawElementProviderSimple>((ComWrappers.ComInterfaceDispatch*)@this).GetPatternProvider(patternId);
+                *ret = obj is null ? null : (void*)AutomationNodeComWrappers.Instance.GetOrCreateComInterfaceForObject(obj, CreateComInterfaceFlags.None);
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                return ex.HResult;
+            }
+        }
+
+        [UnmanagedCallersOnly]
+        public static int GetPropertyValue(void* @this, int propertyId, void** ret)
```

Comment:
This signature is incorrect. What exactly were you using as the source for interface declarations?

Reply:
I'm taking the interface declaration from the reference source https://referencesource.microsoft.com/#UIAutomationProvider/System/Windows/Automation/Provider/IRawElementProviderSimple.cs,067a63891d8bc160
And here I need to implement variant marshalling as well. (that's why I'm still keeping this PR as draft)

Reply:
The underlying type is VARIANT, not IUnknown, so GetOrCreateComInterfaceForObject is incorrect.